plugins/woocommerce/src/Blocks/Utils/BlockHooksTrait.php

```diff
@@ -32,36 +32,39 @@ public function register_hooked_block( $hooked_blocks, $position, $anchor_block,
 			$active_theme_name = wp_get_theme()->get( 'Name' );
 		}
 
-		/**
-		 * A list of pattern slugs to exclude from auto-insert (useful when
-		 * there are patterns that have a very specific location for the block)
-		 *
-		 * @since $VID:$
-		 */
-		$pattern_exclude_list = apply_filters( 'woocommerce_hooked_blocks_pattern_exclude_list', array( 'twentytwentytwo/header-centered-logo', 'twentytwentytwo/header-stacked' ) );
-
 		/**
 		 * A list of theme slugs to execute this with. This is a temporary
 		 * measure until improvements to the Block Hooks API allow for exposing
 		 * to all block themes.
 		 *
-		 * @since $VID:$
+		 * @since 8.5.0
 		 */
 		$theme_include_list = apply_filters( 'woocommerce_hooked_blocks_theme_include_list', array( 'Twenty Twenty-Four', 'Twenty Twenty-Three', 'Twenty Twenty-Two', 'Tsubaki', 'Zaino', 'Thriving Artist', 'Amulet', 'Tazza' ) );
-
 		if ( $context && in_array( $active_theme_name, $theme_include_list, true ) ) {
 			foreach ( $this->hooked_block_placements as $placement ) {
-				if (
-					$placement['position'] === $position &&
-					$placement['anchor'] === $anchor_block &&
-					(
+
+				if ( $placement['position'] === $position && $placement['anchor'] === $anchor_block ) {
+					// If an area has been specified for this placement.
+					if (
 						isset( $placement['area'] ) &&
-						$this->is_template_part_or_pattern( $context, $placement['area'] )
-					) &&
-					! $this->pattern_is_excluded( $context, $pattern_exclude_list ) &&
-					! $this->has_block_in_content( $context )
-				) {
-					$hooked_blocks[] = $this->namespace . '/' . $this->block_name;
+						! $this->has_block_in_content( $context )
+						&& $this->is_target_area( $context, $placement['area'] )
+					) {
+						$hooked_blocks[] = $this->namespace . '/' . $this->block_name;
+					}
+
+					// If no area has been specified for this placement just insert the block.
+					// This is likely to be the case when we're inserting into the navigation block
+					// where we don't have a specific area to target.
+					if ( ! isset( $placement['area'] ) ) {
+						$hooked_blocks[] = $this->namespace . '/' . $this->block_name;
+					}
+
+					// If a callback has been specified for this placement, call it.
+					$callback = isset( $placement['callback'] ) && is_callable( array( $this, $placement['callback'] ) ) ? array( $this, $placement['callback'] ) : null;
+					if ( null !== $callback ) {
+						$hooked_blocks = $callback( $hooked_blocks, $position, $anchor_block, $context );
+					}
```

Comment:
I could change this to become a filter but not sure if I really should as that'll expose it publicly?

Reply:
Yea, can, be added later if the need is demonstrated.